my_package oci

mkrunic-windriver opened the pull request:
Added all necessary files and explanation how to create a my_package.oci VxWorks container image.

Commit:
#1 Added my_package container description and files.

Commit:
#1 Cleaning up README.md.

Commit:
#1 Added reference to container instructions within the main README.md

Commit:
#1 Added missing tag.

Commit:
#1 Added missing tag.

razr commented:
Thanks for the PR! Could you please move it under `examples/vxc` instead of `Docker`, and `create_my_package_container.sh` under `scripts` while renaming it to `create_container.sh`. I don't think this script is `my_package` specific.

```ascii
├── examples
│   └── vxc
│      ├── my_package - VxWorks ROS2 container examples
│         ├── Dockerfile
│         └── README.md
├── scripts
│   └── create_vxc_container.sh
```

Commit:
#1 Folder structure refactored according to PR comment.

Commit:
#1 Renamed script to more precise name.

mkrunic-windriver commented:
I have refactored folder structure according to the review comment.  Regards "Does this script really create a container as it is stated in its name", the answer is it creates [container payload](https://docs.windriver.com/bundle/vxworks_container_programmers_guide_23_03/page/orf1603893608622.html), so I have renamed it to be more precise.

Commit:
#1 Fixed paths in the README.md.

razr commented on README.md:
```diff
@@ -506,6 +506,10 @@ Process 'python3' (process Id = 0xffff80000036cb10) launched.
 
+5. Create `my_package` VxWorks 7 OCI container image
+
+Follow the instructions [here](/Docker/VxWorks7/my_package/README.md)
```
a link shall be fixed

razr commented on scripts/create_vxc_container_payload.sh:
```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+
```
Copyright shall be here

razr commented on scripts/create_vxc_container_payload.sh:
```diff
@@ -0,0 +1,65 @@
+# Validate number of arguments
+if [ $# -lt 4 ] || [ $# -gt 5 ]; then
+    echo "Usage: $0 <SDK_DIR> <OCI_IMG_DIR> <VXC_ARCH> <TOOL> [PACKAGE_PATH]"
+    echo "   eg. $0 /opt/wrsdk/wrsdk-vxworks7-raspberrypi4b-1.2 helloworld.build arm64 llvm"
```
`e.g.,`
as described in the README..md the SDK patch is `~/Downloads/wrsdk` and the package name is `my_package.build`
Is it not better to use default values for this example?
`SDK_DIR=${1:-"~/Downloads/wrsdk"}`
OCI_IMG_DIR=${2:-"./my_package.build"}`
``

Commit:
#1 Cleaning the script and documentation according to PR review.

mkrunic-windriver commented:
I have addressed all points. Please take a look. Thanks for findings.

mkrunic-windriver committed:
Merge branch 'Wind-River:master' into my_package_oci
